force-app/main/default/classes/AN_AutoNumberService.cls

```diff
@@ -202,9 +202,11 @@ public class AN_AutoNumberService {
         }
 
         public String getFormattedNumber(Integer num) {
-            return String.format(displayFormat,
+            Integer numberLength = getNumberLength();
+            String singleDigitDisplayFormat = displayFormat.replace('{000}', '{0}');
```

Comment:
This will fix unit tests, but it wouldn't handle customers who might choose a different format such as "{00000}". You can make it more dynamic by using something like:
```
String singleDigit = '0';
String currentDigitFormat = '{'  + singleDigit.repeat(numberLength) + '}';
String singleDigitDisplayFormat = displayFormat.replace(currentDigitFormat, '{0}');
```

Reply:
I was afraid of something like this, it felt too specific, fix incoming.